src/autoreplace_cmd.cpp

```diff
@@ -412,11 +409,15 @@ static CommandCost CopyHeadSpecificThings(Vehicle *old_head, Vehicle *new_head,
 		if (cost.Succeeded()) cost.AddCost(CmdStartStopVehicle(new_head, false));
 	}
 
-	/* Last do those things which do never fail (resp. we do not care about), but which are not undo-able */
+	/* Do those things which do never fail (resp. we do not care about), but which are not undo-able */
 	if (cost.Succeeded() && old_head != new_head && (flags & DC_EXEC) != 0) {
 		/* Copy other things which cannot be copied by a command and which shall not stay resetted from the build vehicle command */
 		new_head->CopyVehicleConfigAndStatistics(old_head);
 
+		/* Copy group membership */
+		cost.AddCost(DoCommand(0, old_head->group_id, new_head->index, DC_EXEC, CMD_ADD_VEHICLE_GROUP));
+		if (cost.Failed()) return cost;
```

Comment:
The code comment over the if says `never fail`, by moving this here you could now have inconsistencies by skipping the calls below.

Reply:
I can move it to below that block

Reply:
I investigated this further, and there was indeed something else happening behind the scenes that resulted in miscalculated last year profits for the all group and the ungrouped group.

I present a new solution: revert these autoreplace switchings to their origin, and instead, cirurgically update the last year profits of the affected groups in `CopyVehicleConfigAndStatistics`. After re-testings, all the affected groups, including also a custom group now compute last year profits correctly.